repair/table_check.cc

```diff
@@ -13,17 +13,21 @@
 namespace repair {
 
 future<table_dropped> table_sync_and_check(replica::database& db, service::migration_manager& mm, const table_id& uuid) {
-    if (mm.use_raft()) {
-        abort_on_expiry aoe(lowres_clock::now() + std::chrono::seconds{10});
-        auto& as = aoe.abort_source();
-        auto sub = mm.get_abort_source().subscribe([&as] () noexcept {
-            if (!as.abort_requested()) {
-                as.request_abort();
-            }
-        });
-
-        // Trigger read barrier to synchronize schema.
-        co_await mm.get_group0_barrier().trigger(as);
+    try {
+        if (mm.use_raft()) {
+            abort_on_expiry aoe(lowres_clock::now() + std::chrono::seconds{10});
+            auto& as = aoe.abort_source();
+            auto sub = mm.get_abort_source().subscribe([&as] () noexcept {
+                if (!as.abort_requested()) {
+                    as.request_abort();
+                }
+            });
+
+            // Trigger read barrier to synchronize schema.
+            co_await mm.get_group0_barrier().trigger(as);
+        }
+    } catch (...) {
```

Comment:
We don't want to swallow any exception here, just some specific exceptions, like the barrier timing out.
For this reason, we also shouldn't enclose all this code in `try..catch`, just the barrier itself.